src/YouTubeDownloader.php

```diff
@@ -118,7 +118,15 @@ public function parsePlayerResponse($player_response, $js_code)
         $parser = new Parser();
 
         try {
-            $formats = $player_response['streamingData']['formats'];
+            $formats = [];
+            if (isset($player_response['streamingData']))
+            {
+                if (isset($player_response['streamingData']['formats']))
+                {
+                    $formats = $player_response['streamingData']['formats'];
+                }
+            }
+            
             $adaptiveFormats = $player_response['streamingData']['adaptiveFormats'];
```

Comment:
this doesn't solve anything, as `$player_response['streamingData']['adaptiveFormats'];` could still not exist by this point. No point in doing those two isset checks then, if you're just going to let this logic go forward

Reply:
If you use PHP > 7.4, you will get this error if it is null.